### PR TITLE
Add type hints to TFPegasusModel

### PR DESCRIPTION
**Pull request by EdAbati**
# What does this PR do?

This adds type annotations to `TFPegasusModel` and `TFPegasusForConditionalGeneration` as part of #16059

## Who can review?

@Rocketknight1

Thanks :)

### CODE REVIEW

File: src/transformers/models/pegasus/modeling_tf_pegasus.py [hunk 1]
```diff
@@ -1361,25 +1362,25 @@ def set_bias(self, value):
...
+        past_key_values: Optional[Tuple[Tuple[Union[np.ndarray, tf.Tensor]]]] = None,
+        inputs_embeds: Optional[Union[np.ndarray, tf.Tensor]] = None,
+        decoder_inputs_embeds: Optional[Union[np.ndarray, tf.Tensor]] = None,
+        use_cache: Optional[bool] = None,
+        output_attentions: Optional[bool] = None,
+        output_hidden_states: Optional[bool] = None,
+        return_dict: Optional[bool] = None,
+        labels: Optional[Union[np.ndarray, tf.Tensor]] = None,
+        training: bool = False,
+    ) -> Union[TFSeq2SeqModelOutput, Tuple[tf.Tensor]]:
```
Comment (Rocketknight1) [inline src/transformers/models/pegasus/modeling_tf_pegasus.py:1383]: Is the return type here `TFSeq2SeqModelOutput` or `TFSeq2SeqLMOutput`?

**Review by Rocketknight1 — APPROVED** [review]
Overall this looks great! I have one question, after which we're good to merge!

File: src/transformers/models/pegasus/modeling_tf_pegasus.py (hunk 1 above)
Comment (EdAbati) [reply src/transformers/models/pegasus/modeling_tf_pegasus.py]: good catch! Fixed it thanks :)

Comment (EdAbati) [reply src/transformers/models/pegasus/modeling_tf_pegasus.py]: good catch! Fixed it thanks :)

**Comment by Rocketknight1** [comment]
Looks perfect now. Thank you!

**Pull request merged**